pkg/agent/datamodel/types.go

```diff
@@ -350,6 +350,18 @@ const (
 	Webhook AuthenticatorType = "webhook"
 )
 
+type BootstrappingMethod string
+
+const (
+	UseArcMsiToMakeCSR        BootstrappingMethod = "UseArcMsiToMakeCSR"
+	UseAzureMsiToMakeCSR      BootstrappingMethod = "UseAzureMsiToMakeCSR"
+	UseArcMsiDirectly         BootstrappingMethod = "UseArcMsiDirectly"
+	UseAzureMsiDirectly       BootstrappingMethod = "UseAzureMsiDirectly"
+	UseSecureTLSBootstrapping BootstrappingMethod = "UseSecureTLSBootstrapping"
+	//nolint:gosec // this is a const string to use in switch statements, not hardcoded credentials
+	UseTLSBootstrapToken BootstrappingMethod = "UseTLSBootstrapToken"
```

Comment:
I would do something like
```go
const (
	BootstrappingMethodArcMSI                 BootstrappingMethod = "ArcMSI"
	BootstrappingMethodArcMSIToMakeCSR        BootstrappingMethod = "ArcMSIToMakeCSR"
	BootstrappingMethodAzureMSI               BootstrappingMethod = "AzureMSI"
	BootstrappingMethodAzureMSIToMakeCSR      BootstrappingMethod = "AzureMSIToMakeCSR"
	BootstrappingMethodSecureTLSBootstrapping BootstrappingMethod = "SecureTLSBootstrapping"
	//nolint:gosec // this is a const string to use in switch statements, not hardcoded credentials
	BootstrappingMethodTLSBootstrapToken BootstrappingMethod = "TLSBootstrapToken"
)
```

Reply:
It's probably not `BootstrappingMethod`, but something like `ClusterAuthMethod`?